File: src/p1007/python/solution.py
from typing import List


class Solution:
    def minDominoRotations(self, A: List[int], B: List[int]) -> int:
        n = len(A)
        for choose in [A[0], B[0]]:
            i, a, b = 0, 0, 0
            while i < n and (A[i] == choose or B[i] == choose):
                if A[i] != choose: a += 1
                if B[i] != choose: b += 1
                if i + 1 == n: return min(a, b)
                i = i + 1
        return -1

File: src/p1007/cpp/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;


class Solution {
public:
    int minDominoRotations(const vector<int> &A, const vector<int> &B) {
        auto n = A.size();
        int i, a, b;
        for (auto choose : {A[0], B[0]}) {
            for (i = a = b = 0; i < n && (A[i] == choose || B[i] == choose); i++) {
                a += (A[i] != choose); b += (B[i] != choose);
                if (i + 1 == n) return min(a, b);
            }
        }
        return -1;
    }
};
